damnit/backend/listener.py

```diff
@@ -147,22 +148,33 @@ def handle_event(self, record, msg: dict, run_data: RunData):
 
         with log_path.open('ab') as logf:
             # Create subprocess to process the run
-            extract_proc = subprocess.Popen([
-                sys.executable, '-m', 'damnit.backend.extract_data',
-                str(proposal), str(run), run_data.value
-            ], cwd=self.context_dir, stdout=logf, stderr=subprocess.STDOUT)
+            cmd = [sys.executable, '-m', 'damnit.backend.extract_data', str(proposal),
+                   str(run), run_data.value]
+
+            if not self.sandbox:
+                cmd.append('--no-sandbox')
+
+            extract_proc = subprocess.Popen(
+                cmd, cwd=self.context_dir, stdout=logf, stderr=subprocess.STDOUT
+            )
         self.extract_procs_queue.put((proposal, run, extract_proc))
 
-def listen():
+def listen(sandbox: bool):
```

Comment:
Hmm yeah, that was my first idea, but it depends on how 'secure' the configuration should be, since if it's in the database then users could enable/disable the sandboxing as easily as we could, unless the db is read only to the DAMNIT user, which would break a lot of things.

Keeping the setting as part of how the process is started means that the user running the listener could have sandboxing on,  while others can still modify the DB/run reprocessing themselves.

I think options are:

1. Make it part of the table with tables as read-only, enforces sandboxing for processes, but breaks user write permissions to the table.
2. Make it part of the table, still read-write, risk of users disabling sandboxing.
3. Some extra configuration file in the proposal directory/for the listener which is read-only to the DAMNIT user. This kind of config may end up being required as part of the move to a central listener anyway.
4. Something set as part of the supervisor config?
5. Leave it as a flag.

With all options ending in "for now" :stuck_out_tongue:

Reply:
I'd say lets go with option 2 for now, and later we can move it to the centralized listener settings (which should be inaccessible by users).